src/test/javascript/spec/service/activate.service.spec.ts

```diff
@@ -1,19 +1,25 @@
 import { ActivateService } from 'app/account/activate/activate.service';
 import { MockHttpService } from '../helpers/mocks/service/mock-http.service';
-import { HttpParams } from '@angular/common/http';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { TestBed } from '@angular/core/testing';
 
 describe('ActivateService', () => {
     let activateService: ActivateService;
-    let httpService: MockHttpService;
+    let httpService: HttpClient;
     let getStub: jest.SpyInstance;
 
     const getURL = 'api/public/activate';
 
     beforeEach(() => {
-        httpService = new MockHttpService();
-        // @ts-ignore
-        activateService = new ActivateService(httpService);
-        getStub = jest.spyOn(httpService, 'get');
+        TestBed.configureTestingModule({
+            providers: [{ provide: HttpClient, useClass: MockHttpService }],
```

Comment:
we should use `provideHttpClient(), provideHttpClientTesting()` for tests [Http Tests Angular](https://angular.dev/guide/http/testing)

Reply:
Agreed, but the focus here was to get the tests running. Unfortunately, many tests still use the old approach or other approaches that should not be used any more. I suggest we convert this in a separate effort.
The first priority is to get the production code in a modern state, then we can focus on the tests